server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.util;
+
+import org.elasticsearch.Assertions;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Map.entry;
+
+public class Maps {
+
+    /**
+     * Concatenates an entry to an immutable map. This method expects there is not already a mapping for the specified key in the map.
```

Comment:
I think "concatenates" is deceptive wording. This method is linear on the number of entries, while concatenates implies it is adding to the existing map and constant time. Can we please:
* make it clear a copy is made of the immutable map in the javadoc
* use a more clear method name that implies a copy is made, maybe something like "copyMapWithNewEntry" or something like that

Reply:
Thanks for the feedback. I pushed a commit for this.